Sources/Orbit/Components/SegmentedSwitch.swift

```diff
@@ -95,65 +109,30 @@ public struct SegmentedSwitch<Selection: Hashable, Content: View>: View {
     }
 }
 
-public extension AccessibilityID {
-    static let segmentedSwitchFirstOption = Self(rawValue: "orbit.segmentedswitch.first")
-    static let segmentedSwitchSecondOption = Self(rawValue: "orbit.segmentedswitch.second")
-}
-
-// MARK: - Types
-
-/// A view builder for constructing `SegmentedSwitch` content.
-@resultBuilder public enum SegmentedSwitchContentBuilder {
+private extension View {
 
-    static let verticalTextPadding: CGFloat = .small // = 44 @ normal text size
+    @ViewBuilder func segmentFrame(bounds: Anchor<CGRect>, in geometry: GeometryProxy) -> some View {
+        let itemWidth = geometry.size.width / 2
 
-    public static func buildBlock(_ first: some View, _ second: some View) -> some View {
-        HStack(spacing: 0) {
-            switchItem {
-                first
-                    .accessibility(.segmentedSwitchFirstOption)
-            }
-
-            TextStrut()
-                .padding(.vertical, Self.verticalTextPadding)
-
-            switchItem {
-                second
-                    .accessibility(.segmentedSwitchSecondOption)
-            }
-        }
-     }
-
-    @ViewBuilder static func switchItem(content: @escaping () -> some View) -> some View {
-        EnvironmentReader(\.segmentSelection) { selection in
-            PreferenceReader(IDPreferenceKey.self) { preference in
-                content()
-                    .frame(maxWidth: .infinity)
-                    .padding(.horizontal, .small)
-                    .padding(.vertical, verticalTextPadding)
-                    .contentShape(Rectangle())
-                    .onTapGesture {
-                        selection.wrappedValue = preference.first?.id
-                    }
-            }
-        }
+        self
+            .frame(width: itemWidth)
+            .offset(x: (geometry[bounds].minX / itemWidth).rounded(.down) * itemWidth)
     }
 }
 
-struct SegmentSelectionKey: EnvironmentKey {
-    static let defaultValue: Binding<AnyHashable?> = .constant(nil)
-}
-
-extension EnvironmentValues {
-    var segmentSelection: Binding<AnyHashable?> {
-        get { self[SegmentSelectionKey.self] }
-        set { self[SegmentSelectionKey.self] = newValue }
-    }
+public extension AccessibilityID {
+    static let segmentedSwitchFirstOption = Self(rawValue: "orbit.segmentedswitch.first")
```

Comment:
These are now unused. With any content, call site needs to provide its own identifiers.

Reply:
The identifier can be checked in SwiftUI previews as well

<img width="725" alt="image" src="https://github.com/kiwicom/orbit-swiftui/assets/35844477/fd515bba-d43b-4bf5-8432-afb8693c9d6a">